### PR TITLE
BAH-2198 | Add script to set artifact version

### PR DESCRIPTION
**Pull request by mohan-13**
This PR adds a shell script that can be used to set artifact version in github actions.

Co-authored-by: stavtnameh <hemanth.v@beehyv.com>

### CODE REVIEW

**Review by mradul1807 — APPROVED** [review]

File: setArtifactVersion.sh [hunk 1]
```diff
@@ -0,0 +1,36 @@
+#!/bin/bash
+set -e
+
+checkIfTagExists(){
```
Comment (N0-man) [inline setArtifactVersion.sh]: `verifyReleaseVersion`

File: setArtifactVersion.sh [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  version=$1
+  tagCount=$(curl -s https://api.github.com/repos/$GITHUB_REPOSITORY/tags | jq --arg tagName "$version"  '[.[] | select( .name == $tagName)] | length')
+  if [ $tagCount -gt 0 ]; then
+    echo "Error: Version $version already released. Please update your version in package/.appversion"
+    exit 1
+  fi
+}
+
+setVersionAsGithubEnv(){
```
Comment (N0-man) [inline setArtifactVersion.sh]: `setArtifactVersion`

File: setArtifactVersion.sh [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+  version=$1
+  echo "Setting version $version"
+  echo "ARTIFACT_VERSION=$version" >> $GITHUB_ENV
+}
+
+case $GITHUB_REF in
+  refs/tags/*)
+      echo "Current action is for tag.."
```
Comment (N0-man) [inline setArtifactVersion.sh:21]: these echo logs are to indicate the action intent - such as `/tags` would create release version, `/heads/release-*` would create release candidate artifact version etc

**Review by N0-man — APPROVED** [review]

File: setArtifactVersion.sh (hunk 3 above)
Comment (mohan-13) [reply setArtifactVersion.sh:21]: Yes Nouman.

**Pull request merged**